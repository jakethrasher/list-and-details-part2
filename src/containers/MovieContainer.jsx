import React, { useState, useEffect } from 'react';
import MovieList from '../components/movies/MovieList';
import { fetchMovies } from '../services/movieDbApi';

const MovieContainer = () =>{
    const [movies, setMovies] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        (async ()=>{
            const movies = await fetchMovies()
            setMovies(movies);
            setLoading(false)
        })()
    }, []);
    
    
    if(loading)return <h1>Loading...</h1>
    else return (
        <>
            <MovieList movies={movies}/>
        </>
    )
}

export default MovieContainer;