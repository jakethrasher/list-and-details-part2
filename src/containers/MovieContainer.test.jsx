import dotenv from 'dotenv';
dotenv.config();
import React from 'react'
import { screen, render, waitFor } from '@testing-library/react'
import MovieContainer from './MovieContainer'
import { MemoryRouter} from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import moviesData from '../../fixtures/moviesData.json';

const server = setupServer(
    rest.get(`https://api.themoviedb.org/3/movie/popular?api_key=${process.env.MOVIE_DATABASE_KEY}&language=en-US`, (req, res, ctx) => {
        return res(ctx.json(
            moviesData
        ))
    })
)
describe('MovieDatabase Container', () => {
    beforeAll(()=> server.listen());
    afterAll(() => server.close());

    it('diplays a list of movies', async () => {
        render(
            <MemoryRouter>
                <MovieContainer/>
            </MemoryRouter>
        )
        screen.getByText('Loading...')
        const ul = await screen.findByRole('list',{ name: 'movies'});
        expect(ul).toMatchSnapshot()
    })
})