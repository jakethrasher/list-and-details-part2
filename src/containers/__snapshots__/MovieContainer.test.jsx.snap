// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDatabase Container diplays a list of movies 1`] = `
<ul
  aria-label="movies"
>
  <a
    class="linkStyle"
    href="/567189"
  >
    <li>
      <figure>
        <img
          alt="Tom Clancy's Without Remorse"
          src="http://image.tmdb.org/t/p/w200//rEm96ib0sPiZBADNKBHKBv5bve9.jpg"
        />
        <figcaption>
          <p>
            Tom Clancy's Without Remorse
          </p>
          <p>
            An elite Navy SEAL uncovers an international conspiracy while seeking justice for the murder of his pregnant wife.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/460465"
  >
    <li>
      <figure>
        <img
          alt="Mortal Kombat"
          src="http://image.tmdb.org/t/p/w200//nkayOAUBUu4mMvyNf9iHSUiPjF1.jpg"
        />
        <figcaption>
          <p>
            Mortal Kombat
          </p>
          <p>
            Washed-up MMA fighter Cole Young, unaware of his heritage, and hunted by Emperor Shang Tsung's best warrior, Sub-Zero, seeks out and trains with Earth's greatest champions as he prepares to stand against the enemies of Outworld in a high stakes battle for the universe.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/399566"
  >
    <li>
      <figure>
        <img
          alt="Godzilla vs. Kong"
          src="http://image.tmdb.org/t/p/w200//pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
        />
        <figcaption>
          <p>
            Godzilla vs. Kong
          </p>
          <p>
            In a time when monsters walk the Earth, humanity’s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/615457"
  >
    <li>
      <figure>
        <img
          alt="Nobody"
          src="http://image.tmdb.org/t/p/w200//oBgWY00bEFeZ9N25wWVyuQddbAo.jpg"
        />
        <figcaption>
          <p>
            Nobody
          </p>
          <p>
            Hutch Mansell, a suburban dad, overlooked husband, nothing neighbor — a "nobody." When two thieves break into his home one night, Hutch's unknown long-simmering rage is ignited and propels him on a brutal path that will uncover dark secrets he fought to leave behind.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/811367"
  >
    <li>
      <figure>
        <img
          alt="22 vs. Earth"
          src="http://image.tmdb.org/t/p/w200//32vLDKSzcCMh55zqqaSqqUA8naz.jpg"
        />
        <figcaption>
          <p>
            22 vs. Earth
          </p>
          <p>
            Set before the events of ‘Soul’, 22 refuses to go to Earth, enlisting a gang of 5 new souls in attempt of rebellion. However, 22’s subversive plot leads to a surprising revelation about the meaning of life.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/813258"
  >
    <li>
      <figure>
        <img
          alt="Monster Pets: A Hotel Transylvania Short"
          src="http://image.tmdb.org/t/p/w200//dkokENeY5Ka30BFgWAqk14mbnGs.jpg"
        />
        <figcaption>
          <p>
            Monster Pets: A Hotel Transylvania Short
          </p>
          <p>
            Drac tries out some new monster pets to help occupy Tinkles for playtime.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/632357"
  >
    <li>
      <figure>
        <img
          alt="The Unholy"
          src="http://image.tmdb.org/t/p/w200//b4gYVcl8pParX8AjkN90iQrWrWO.jpg"
        />
        <figcaption>
          <p>
            The Unholy
          </p>
          <p>
            Alice, a young hearing-impaired girl who, after a supposed visitation from the Virgin Mary, is inexplicably able to hear, speak and heal the sick. As word spreads and people from near and far flock to witness her miracles, a disgraced journalist hoping to revive his career visits the small New England town to investigate. When terrifying events begin to happen all around, he starts to question if these phenomena are the works of the Virgin Mary or something much more sinister.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/791373"
  >
    <li>
      <figure>
        <img
          alt="Zack Snyder's Justice League"
          src="http://image.tmdb.org/t/p/w200//tnAuB8q5vv7Ax9UAEje5Xi4BXik.jpg"
        />
        <figcaption>
          <p>
            Zack Snyder's Justice League
          </p>
          <p>
            Determined to ensure Superman's ultimate sacrifice was not in vain, Bruce Wayne aligns forces with Diana Prince with plans to recruit a team of metahumans to protect the world from an approaching threat of catastrophic proportions.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/726684"
  >
    <li>
      <figure>
        <img
          alt="Miraculous World Shanghai, la légende de Ladydragon"
          src="http://image.tmdb.org/t/p/w200//msI5a9TPnepx47JUb2vl88hb80R.jpg"
        />
        <figcaption>
          <p>
            Miraculous World Shanghai, la légende de Ladydragon
          </p>
          <p>
            To join Adrien in Shanghai, Marinette is going to visit her uncle Wang who is celebrating his anniversary. But, as soon as she arrives in China, her purse gets stolen with Tikki inside, whom she needs to secretly transform into Ladybug! Without money and alone in the immense city, Marinette accepts the help of a young and resourceful girl, Fei. The two girls will ally and discover the existence of a new magical jewel, the Prodigious. Hawk Moth, also present in Shanghai, seeks to finding it since a long time...
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/635302"
  >
    <li>
      <figure>
        <img
          alt="劇場版「鬼滅の刃」無限列車編"
          src="http://image.tmdb.org/t/p/w200//h8Rb9gBr48ODIwYUttZNYeMWeUU.jpg"
        />
        <figcaption>
          <p>
            劇場版「鬼滅の刃」無限列車編
          </p>
          <p>
            Tanjirō Kamado, joined with Inosuke Hashibira, a boy raised by boars who wears a boar's head, and Zenitsu Agatsuma, a scared boy who reveals his true power when he sleeps, boards the Infinity Train on a new mission with the Fire Hashira, Kyōjurō Rengoku, to defeat a demon who has been tormenting the people and killing the demon slayers who oppose it!
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/804435"
  >
    <li>
      <figure>
        <img
          alt="Vanquish"
          src="http://image.tmdb.org/t/p/w200//AoWY1gkcNzabh229Icboa1Ff0BM.jpg"
        />
        <figcaption>
          <p>
            Vanquish
          </p>
          <p>
            Victoria is a young mother trying to put her dark past as a Russian drug courier behind her, but retired cop Damon forces Victoria to do his bidding by holding her daughter hostage. Now, Victoria must use guns, guts and a motorcycle to take out a series of violent gangsters—or she may never see her child again.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/527774"
  >
    <li>
      <figure>
        <img
          alt="Raya and the Last Dragon"
          src="http://image.tmdb.org/t/p/w200//lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg"
        />
        <figcaption>
          <p>
            Raya and the Last Dragon
          </p>
          <p>
            Long ago, in the fantasy world of Kumandra, humans and dragons lived together in harmony. But when an evil force threatened the land, the dragons sacrificed themselves to save humanity. Now, 500 years later, that same evil has returned and it’s up to a lone warrior, Raya, to track down the legendary last dragon to restore the fractured land and its divided people.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/615678"
  >
    <li>
      <figure>
        <img
          alt="Thunder Force"
          src="http://image.tmdb.org/t/p/w200//3mKMWP5OokB7QpcOMA1yl8BXFAF.jpg"
        />
        <figcaption>
          <p>
            Thunder Force
          </p>
          <p>
            In a world where supervillains are commonplace, two estranged childhood best friends reunite after one devises a treatment that gives them powers to protect their city.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/643586"
  >
    <li>
      <figure>
        <img
          alt="Willy's Wonderland"
          src="http://image.tmdb.org/t/p/w200//keEnkeAvifw8NSEC4f6WsqeLJgF.jpg"
        />
        <figcaption>
          <p>
            Willy's Wonderland
          </p>
          <p>
            When his car breaks down, a quiet loner agrees to clean an abandoned family fun center in exchange for repairs. He soon finds himself waging war against possessed animatronic mascots while trapped inside Willy's Wonderland.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/681260"
  >
    <li>
      <figure>
        <img
          alt="Maya the Bee: The Golden Orb"
          src="http://image.tmdb.org/t/p/w200//tMS2qcbhbkFpcwLnbUE9o9IK4HH.jpg"
        />
        <figcaption>
          <p>
            Maya the Bee: The Golden Orb
          </p>
          <p>
            When Maya, a headstrong little bee, and her best friend Willi, rescue an ant princess they find themselves in the middle of an epic bug battle that will take them to strange new worlds and test their friendship to its limits.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/634528"
  >
    <li>
      <figure>
        <img
          alt="The Marksman"
          src="http://image.tmdb.org/t/p/w200//6vcDalR50RWa309vBH1NLmG2rjQ.jpg"
        />
        <figcaption>
          <p>
            The Marksman
          </p>
          <p>
            Jim Hanson’s quiet life is suddenly disturbed by two people crossing the US/Mexico border – a woman and her young son – desperate to flee a Mexican cartel. After a shootout leaves the mother dead, Jim becomes the boy’s reluctant defender. He embraces his role as Miguel’s protector and will stop at nothing to get him to safety, as they go on the run from the relentless assassins.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/412656"
  >
    <li>
      <figure>
        <img
          alt="Chaos Walking"
          src="http://image.tmdb.org/t/p/w200//9kg73Mg8WJKlB9Y2SAJzeDKAnuB.jpg"
        />
        <figcaption>
          <p>
            Chaos Walking
          </p>
          <p>
            Two unlikely companions embark on a perilous adventure through the badlands of an unexplored planet as they try to escape a dangerous and disorienting reality, where all inner thoughts are seen and heard by everyone.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/458576"
  >
    <li>
      <figure>
        <img
          alt="Monster Hunter"
          src="http://image.tmdb.org/t/p/w200//1UCOF11QCw8kcqvce8LKOO6pimh.jpg"
        />
        <figcaption>
          <p>
            Monster Hunter
          </p>
          <p>
            A portal transports Cpt. Artemis and an elite unit of soldiers to a strange world where powerful monsters rule with deadly ferocity. Faced with relentless danger, the team encounters a mysterious hunter who may be their only hope to find a way home.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/464052"
  >
    <li>
      <figure>
        <img
          alt="Wonder Woman 1984"
          src="http://image.tmdb.org/t/p/w200//8UlWHLMpgZm9bx6QYh0NFoq67TZ.jpg"
        />
        <figcaption>
          <p>
            Wonder Woman 1984
          </p>
          <p>
            A botched store robbery places Wonder Woman in a global battle against a powerful and mysterious ancient force that puts her powers in jeopardy.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
  <a
    class="linkStyle"
    href="/578701"
  >
    <li>
      <figure>
        <img
          alt="Those Who Wish Me Dead"
          src="http://image.tmdb.org/t/p/w200//xCEg6KowNISWvMh8GvPSxtdf9TO.jpg"
        />
        <figcaption>
          <p>
            Those Who Wish Me Dead
          </p>
          <p>
            A young boy finds himself pursued by two assassins in the Montana wilderness with a survival expert determined to protecting him - and a forest fire threatening to consume them all.
          </p>
        </figcaption>
      </figure>
    </li>
  </a>
</ul>
`;
